enterprise/backend/test/metabase_enterprise/sandbox/query_processor/middleware/row_level_restrictions_test.clj

```diff
@@ -354,15 +358,16 @@
                   "card) to see if row level permissions apply. This was broken when it wasn't expecting a card and "
                   "only expecting resolved source-tables")
       (mt/with-temp Card [card {:dataset_query (mt/mbql-query venues)}]
-        (mt/with-test-user :rasta
-          (is (= [[100]]
-                 (mt/format-rows-by [int]
-                   (mt/rows
-                    (qp/process-query
-                     {:database (mt/id)
-                      :type     :query
-                      :query    {:source-table (format "card__%s" (u/the-id card))
-                                 :aggregation  [["count"]]}}))))))))))
+        (let [query {:database (mt/id)
+                     :type     :query
+                     :query    {:source-table (format "card__%s" (u/the-id card))
+                                :aggregation  [["count"]]}}]
+          (mt/with-test-user :rasta
+            (mt/with-native-query-testing-context query
```

Comment:
It would be nice if this was a binding macro that allowed you to define `query`.